tfhe/src/integer/server_key/radix/comparison.rs

```diff
@@ -82,12 +83,14 @@ impl ServerKey {
                 .collect::<Vec<_>>();
         }
 
-        block_comparisons.resize_with(lhs.blocks().len(), || self.key.create_trivial(0));
-
-        T::from_blocks(block_comparisons)
+        let result = block_comparisons
+            .into_iter()
+            .next()
+            .unwrap_or_else(|| self.key.create_trivial(1));
```

Comment:
why is the default value true here ?

the into iter next fails if block_comparisons are empty i.e. if the LHS is empty 🤔 I've seen the pattern elsewhere, maybe we should not set a default value with the unwrap_or_else but rather expect with an internal error message ?

(same where this pattern is used elsewhere, there are a few places) unless I'm missing something

Reply:
It was to handle the case when inputs are empty, if both lhs and rhs are emtpy then they are equal.

but actually it will also return the trivial true if lhs is empty but rhs is not, which is not correct.
Maybe its better to assert lhs.len() == rhs.len(), and at the begining

Reply:
yep